diff --git a/lib/libv4l-gst/gst-backend.c b/lib/libv4l-gst/gst-backend.c
index 8684fc4..d8c47b2 100644
--- a/lib/libv4l-gst/gst-backend.c
+++ b/lib/libv4l-gst/gst-backend.c
@@ -2915,26 +2915,48 @@ unlock:
 
 	return map;
 }
+
 int expbuf_ioctl(struct v4l_gst_priv *dev_ops_priv,
 		 struct v4l2_exportbuffer *expbuf) {
 	struct v4l_gst_buffer *buffer;
 	struct gst_backend_priv *priv = dev_ops_priv->gst_priv;
-	guint i;
-	if (expbuf->type != V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE) {
-		fprintf(stderr, "Can only export capture buffers as dmebuf\n");
+	guint i = 0;
+	GstMemory *mem = NULL;
+
+	if ((expbuf->type == V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE) ||
+	    (expbuf->type == V4L2_BUF_TYPE_PRIVATE)) {
+		buffer = &priv->cap_buffers[expbuf->index];
+		if (expbuf->plane < gst_buffer_n_memory(buffer->buffer)) {
+			i = expbuf->plane;
+		}
+
+		mem = gst_buffer_peek_memory(buffer->buffer, i);
+		if (!gst_is_dmabuf_memory(mem)) {
+			fprintf(stderr, "Failed to get dambuf emmory.\n");
+			return -1;
+		}
+	}
+
+	if (mem == NULL) {
+		fprintf(stderr, "Invalid type.\n");
 		errno = EINVAL;
 		return -1;
 	}
-	buffer = &priv->cap_buffers[expbuf->index];
-	for (i = 0; i < gst_buffer_n_memory(buffer->buffer); i++) {
-		GstMemory *mem;
-		mem = gst_buffer_peek_memory(buffer->buffer, i);
-		if (!gst_is_dmabuf_memory(mem))
-			continue;
+
+	switch(expbuf->type) {
+	case V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE:
 		expbuf->fd = dup(gst_dmabuf_memory_get_fd (mem));
-		return 0;
+		break;
+	case V4L2_BUF_TYPE_PRIVATE:
+		expbuf->reserved[0] = mem->offset;
+		break;
+	default:
+		fprintf(stderr, "Can only export capture buffers as dmebuf\n");
+		errno = EINVAL;
+		return -1;
 	}
-	return -1;
+
+	return 0;
 }
 
 int g_selection_ioctl(struct v4l_gst_priv *dev_ops_priv, struct v4l2_selection *selection) {
